be1-go/internal/popserver/config/config.go

```diff
@@ -0,0 +1,102 @@
+package config
+
+import (
+	"encoding/base64"
+	"go.dedis.ch/kyber/v3"
+	"golang.org/x/xerrors"
+	"popstellar/message/answer"
+	"sync"
+	"testing"
+)
+
+var once sync.Once
+var instance *config
+
+type config struct {
+	ownerPubKey         kyber.Point
+	serverPubKey        kyber.Point
+	serverSecretKey     kyber.Scalar
+	clientServerAddress string
+	serverServerAddress string
+}
+
+func InitConfig(ownerPubKey, serverPubKey kyber.Point, serverSecretKey kyber.Scalar,
+	clientServerAddress, serverServerAddress string) {
+	once.Do(func() {
+		instance = &config{
+			ownerPubKey:         ownerPubKey,
+			serverPubKey:        serverPubKey,
+			serverSecretKey:     serverSecretKey,
+			clientServerAddress: clientServerAddress,
+			serverServerAddress: serverServerAddress,
+		}
+	})
+}
+
+func SetConfig(t *testing.T, ownerPubKey, serverPubKey kyber.Point, serverSecretKey kyber.Scalar,
```

Comment:
I recommend against using the `testing` package in production code. This is a recurrent issue. The only solution I see that won't require a lot of refactoring from your part is to have regular setters and only use them in tests (without passing a testing variable).

Reply:
I made the change that you wanted for the setters with a big comment above to only use inside test. But for the generator I just renamed it generatortest as we only use those generators in the purpose of testing and will need those function later to do some end to end test.